### PR TITLE
add download_policy to katello_repository

### PR DESCRIPTION
**Pull request by mdellweg**
Sadly nailgun allows this write-only...

### CODE REVIEW

**Comment by mdellweg** [comment]
@ehelms thoughts?

File: modules/katello_repository.py [hunk 1]
```diff
@@ -130,8 +140,10 @@ def repository(self, name, content_type, product, organization, url=None):
 
         repository = repository[0] if len(repository) == 1 else None
 
-        if repository and (repository.name != name or repository.url != url):
-            repository = self._entities.Repository(self._server, name=name, id=repository.id, url=url)
+        # Sadly, nailgun does not handle 'download_policy' as we would like it.
```
Comment (ehelms) [inline modules/katello_repository.py]: Does this imply it currently only works on creation? Do we have an issue open to nailgun for this?

Comment (mdellweg) [reply modules/katello_repository.py]: If i remember correctly, it is write-only. So creating and updating works, but reading (which is important for 'check_mode' and nice in any case) does not. Haven't had the time to get this to nailgun yet.

Comment (mdellweg) [reply modules/katello_repository.py]: The only thing @ nailgun i can find right now is not in repositories, but in smart-proxy. Reinvestigating this right now.

Comment (mdellweg) [reply modules/katello_repository.py]: Nailgun was not to blame here. Sorry!
A missing 'read()' was.

**Comment by mdellweg** [comment]
@ehelms and rebased...

File: modules/katello_repository.py (hunk 1 above)
Comment (ehelms) [reply modules/katello_repository.py]: Awesome!

**Pull request merged**